.github/workflows/python-app.yml

```diff
@@ -83,7 +83,6 @@ jobs:
 
   dist:
     runs-on: ubuntu-latest
-    needs: [coverage]
```

Comment:
now that i think of it, this is also removing the test run as dependency, no? we still want that.

Reply:
I did a refactor to workflows (because I wanted to leave async tests as the check for dist instead of coverage, because coverage is much slower). Is also for reorganization, let me know what you think.